sentry/src/main/java/io/sentry/SentryTracer.java

```diff
@@ -150,8 +188,9 @@ public void finish() {
 
   @Override
   public void finish(@Nullable SpanStatus status) {
-    if (!root.isFinished()) {
-      root.finish(status);
+    this.finishStatus = status;
+    if (!root.isFinished() && (!waitForChildren || hasAllChildrenFinished())) {
```

Comment:
I'm missing a finish intention flag here.
Let's say the user did not call finish, but finishes all children, this condition will evaluate to true and finish the root span.

Reply:
IOS solved this by adding a ```isWaitingForChildren``` variable, and a private method ```canBeFinished```.
The finish method only sets isWaitingForChildren to true, saves the status and call canBeFinished that does the rest. 
The span listener should call ```canBeFinished```, otherwise it would set the isWaitingForChildren to true.

Reply:
I think its fixed now: https://github.com/getsentry/sentry-java/pull/1535/files#diff-8bb325ca6fc9666c24826415b357d8e76a34cb455edc5709dd962bdcd34c6f16R136

Reply:
Nice!
You just fixed while I was commenting 😂

Reply:
its fixed, thanks!